packages/DropdownMenu/src/DropdownMenu.js

```diff
@@ -0,0 +1,102 @@
+import React from "react";
+import PropTypes from "prop-types";
+import Popover from "@paprika/popover";
+import ContentContainerStyled from "./ContentContainer.styles";
+import Confirmation from "./Confirmation/Confirmation";
+import Item from "./Item/Item";
+
+const { alignTypes, oneOf, node, string, func } = PropTypes;
+
+const propTypes = {
+  /** Alignment of the dropdown menu */
+  align: oneOf(alignTypes),
+
+  /** Children should consist of <Dropdown.Item /> */
+  children: node.isRequired,
+
+  /** Optional custom classes */
+  className: string,
+
+  /** Render prop for rendering the trigger element that toggles the dropdown */
+  renderTrigger: func.isRequired,
+};
+
+const defaultProps = {
+  align: "bottom",
+  className: "",
+};
+
+const DropDownMenu = props => {
+  const { align, children, className } = props;
+  const [isOpen, setIsOpen] = React.useState();
+  const [isConfirming, setIsConfirming] = React.useState();
+  const [renderConfirmation, setRenderConfirmation] = React.useState();
+
+  const handleCloseMenu = () => {
+    setIsOpen(false);
+    setIsConfirming(false);
+    setRenderConfirmation(null);
+  };
+
+  const handleOpenMenu = () => {
+    setIsOpen(true);
+  };
+
+  const handleShowConfirmation = renderConfirmation => () => {
+    setIsConfirming(prevIsConfirmingState => !prevIsConfirmingState);
+    setRenderConfirmation(prevIsConfirmingState => (prevIsConfirmingState ? null : renderConfirmation));
+  };
+
+  const renderContent = () => {
+    if (isConfirming) {
+      return renderConfirmation(handleCloseMenu);
+    }
+
+    return (
+      <ContentContainerStyled isOpen={isOpen}>
+        {React.Children.toArray(children).map(child => {
+          if (child.type.componentType === "DropDownMenu.Item") {
+            if (child.props.renderConfirmation) {
+              return React.cloneElement(child, {
+                onShowConfirmation: handleShowConfirmation(child.props.renderConfirmation),
+              });
+            }
+            return React.cloneElement(child, {
+              onClose: () => {
+                handleCloseMenu();
+              },
+            });
+          }
+
+          return child;
+        })}
+      </ContentContainerStyled>
+    );
+  };
+
+  return (
+    <Popover
+      align={align}
+      className={className}
+      offset={4}
+      role={!isConfirming ? "menu" : null}
+      isOpen={isOpen}
+      onClose={() => {
+        if (!isConfirming) {
+          handleCloseMenu();
+        }
```

Comment:
onClose is being triggered by the Paprika popover component when we want it to render the confirmation panel. This is because of the unmount of rendered content in the popover content causing an `onBlur` event. 
Another solution may be to ensure focus is retained on the popover content so as not to trigger the onBlur, and to stop an `onClose` event, similar to how we do in the Date Picker component - https://github.com/acl-services/paprika/pull/94/files#diff-f2f7e73e1aac4537ad454c8d3e6c3258R63